File: problemas/iniciante/1557/1557.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

let digitos = (numero) => Math.floor(Math.log10(numero)) + 1;

while (lines.length) {
    let N = parseInt(lines.shift());

    if (N === 0) {
        break;
    }

    let limite = digitos(1 << (2 * N - 2));

    for (let i = 0; i < N; ++i) {
        let linha = [];
        linha.push((1 << i).toString().padStart(limite));
        for (let j = 1; j < N; ++j) {
            linha.push((1 << (i + j)).toString().padStart(limite));
        }
        console.log(linha.join(' '));
    }

    console.log('');
}

File: problemas/iniciante/1557/1557.py
import math

def digitos(numero):
  return math.floor(math.log10(numero)) + 1

while True:
  try:
    N = int(input())

    if (N == 0):
      break

    limite = digitos(1 << (2 * N - 2))

    for i in range(N):
      print(str(1 << i).rjust(limite), end='')
      for j in range(1, N):
        print(' ', end='')
        print(str(1 << (i + j)).rjust(limite), end='')
      print('')
    print('')
  except EOFError:
    break
